sky/services/dynamic/dynamic_service_dylib.cc

```diff
@@ -0,0 +1,47 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "sky/services/dynamic/dynamic_service.h"
+
+#include <cassert>
+
+#include "sky/services/dynamic/dynamic_service_dylib.h"
+#include "mojo/public/cpp/environment/environment.h"
+
+static mojo::Environment* g_dylib_environment;
+
+void FlutterServiceOnLoad(const MojoAsyncWaiter* waiter,
+                          const MojoLogger* logger) {
+  // Assert that the dylib environment is not already initialized somehow.
+  assert(g_dylib_environment == nullptr);
+  g_dylib_environment = new mojo::Environment(waiter, logger);
+}
+
+void FlutterServiceInvoke(MojoHandle client_handle, const char* service_name) {
+  // The service is always responsible for releasing the client handle. Create
+  // a scoped handle immediately.
+  mojo::MessagePipeHandle message_pipe_handle(client_handle);
+  mojo::ScopedMessagePipeHandle scoped_handle(message_pipe_handle);
+
+  if (g_dylib_environment == nullptr) {
+    return;
+  }
+
+  mojo::String name(service_name);
+
+  if (name.size() == 0) {
+    return;
```

Comment:
the service name comes from the flutter engine, right? if the engine passes bogus data in i'm not sure you want the dylib to have to try to deal with it, probably better to have all of the validation happen on the engine side